unified-runtime/source/adapters/native_cpu/enqueue.cpp

```diff
@@ -112,6 +101,13 @@ UR_APIEXPORT ur_result_t UR_APICALL urEnqueueKernelLaunch(
   // TODO: add proper error checking
   native_cpu::NDRDescT ndr(workDim, pGlobalWorkOffset, pGlobalWorkSize,
                            pLocalWorkSize);
+  unsigned long long numWI;
+  if (__builtin_umulll_overflow(ndr.GlobalSize[0], ndr.GlobalSize[1], &numWI) ||
+      __builtin_umulll_overflow(numWI, ndr.GlobalSize[2], &numWI) ||
```

Comment:
I think we build this with the host compiler, in which case this would presumably fail on MSVC, but that is not currently being tested. I'll change this to something portable.

Reply:
Done.